server/src/utils/asset.util.ts

```diff
@@ -0,0 +1,91 @@
+import { AccessCore, Permission } from 'src/cores/access.core';
+import { BulkIdErrorReason, BulkIdResponseDto } from 'src/dtos/asset-ids.response.dto';
+import { AuthDto } from 'src/dtos/auth.dto';
+import { IAccessRepository } from 'src/interfaces/access.interface';
+import { setDifference, setUnion } from 'src/utils/set';
+
+export interface IBulkAsset {
+  getAssetIds: (id: string, assetIds: string[]) => Promise<Set<string>>;
+  addAssetIds: (id: string, assetIds: string[]) => Promise<void>;
+  removeAssetIds: (id: string, assetIds: string[]) => Promise<void>;
+}
+
+export const addAssets = async (
+  auth: AuthDto,
+  repositories: { accessRepository: IAccessRepository; repository: IBulkAsset },
+  dto: { id: string; assetIds: string[] },
+) => {
+  const { accessRepository, repository } = repositories;
```

Comment:
Nit: you can destructure the parameter directly.

Reply:
I know, but I didn't want to cluster the parameters even more lol

Reply:
You would prefer destructuring them directly?

Reply:
Personal preference! I only destructure at the top of a function body if doing it in the parameter will blow it up to multiple lines.